clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -698,8 +698,10 @@ scheduler_tmpl_t<RecordType, ReaderType>::advance_region_of_interest(int output_
 {
     uint64_t cur_instr = input.reader->get_instruction_ordinal();
     auto &cur_range = input.regions_of_interest[input.cur_region];
```

Comment:
Assert that input.cur_region is not out of bounds.

Reply:
Done.